lib/ruby/lib/alces/stack/boot/run.rb

```diff
@@ -163,9 +182,11 @@ def teardown(e)
           tear_down_flag = false
           tear_down_flag_dry = true if @dry_run_flag and !@to_delete.empty?
           tear_down_flag = true if !@dry_run_flag and !@to_delete_dry_run.empty?
-          puts "DRY RUN: Files that would be deleted:" if !@to_delete_dry_run.empty?
-          @to_delete_dry_run.each do |file| puts "  #{file}" end
-          @to_delete.each do |file| `rm -f #{file} 2>/dev/null` end
+          if !@permanent_boot_flag
+            STDERR.puts "DRY RUN: Files that would be deleted:" if !@to_delete_dry_run.empty?
+            @to_delete_dry_run.each do |file| STDERR.puts "  #{file}" end
```

Comment:
Prefer `{...}` form for blocks when they're one-liners; ref: https://github.com/bbatsov/ruby-style-guide#single-line-blocks